### ISSUE TITLE
Exclude unexpected behavior 

### ISSUE THREAD

**Issue by yohai7700** [microsoft/TypeScript#44901]
# Bug Report

### 🔎 Search Terms
"Exclude"
"Exclude shallow"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about Generic

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.0-dev.20210705#code/C4TwDgpgBAcg9gOwIIGVgCcCWCDmUC8UAogB4DGANgK4AmEAPAM4bY4A0UA5AIacB8AbgBQZRMygkAXLESoWuAlABE3JcKGhIUAEJosCwqUq0GPTlAA+XAEbmOZwSLHAoIabvl5CKtUA)

### 💻 Code
```ts
type NonAString = Exclude<string, 'a'>;
const x: NonAString = "a";

type BString = Exclude<'a' | 'b' , 'a'>;
const y: BString = "a";
```

### 🙁 Actual behavior
```
const x: NonAString = "a";
```
Doesn't throw an error, although 'a' is excluded so it should not be able to be assigned.

unlike,
```
const y: BString = "a";
```
that throws a `Type '"a"' is not assignable to type '"b"'.` error.

### 🙂 Expected behavior

`Type '"a"' is not assignable to type "Exclude<string, 'a'>".`
because Exclude documentation is: "Exclude from T those types that are assignable to U", so you should only be able to assign non-"a" strings to `Exclude<string, 'a'>`.

**Comment by MartinJohns** [comment]
Exclude only excludes one type from a union of types. String is no union. There is no way in the type system to represent a type "any string, except this one". This would require something like #29317.

**Comment by yohai7700** [comment]
> Exclude only excludes one type from a union of types. String is no union. There is no way in the type system to represent a type "any string, except this one". This would require something like #29317.

ok. thanks for the answer.
